File: 119.pascals-triangle-ii.py
#
# @lc app=leetcode id=119 lang=python3
#
# [119] Pascal's Triangle II
#

# @lc code=start
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        res = []

        for i in range(rowIndex + 1):
            for j in range(i - 1, 0, -1):
                res[j] = res[j] + res[j - 1]
            res.append(1)
        return res
                



    
        
# @lc code=end

File: 119.pascals-triangle-ii.java
/*
 * @lc app=leetcode id=119 lang=java
 *
 * [119] Pascal's Triangle II
 */

// @lc code=start

import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> res = new ArrayList<>();

        for(int i = 0; i < rowIndex + 1; i++) {
            for (int j = i - 1; j > 0; j--) {
                res.set(j, res.get(j) + res.get(j - 1));
            }
            res.add(1);
        }
        return res;
    }
}
// @lc code=end

